dagshub/storage/rclone.py

```diff
@@ -0,0 +1,238 @@
+                              universal_newlines=True) as p:
+            for line in p.stdout:
+                print(line, end='')
+
+        print(f"Successfully synchronized {local_path_str} to DagsHub Storage {remote_path}.")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to synchronize {local_path_str} to DagsHub Storage {remote_path}: {e}")
+
+
+def mount_bucket(repo: str, cache: bool = False, path: Path = None):
+    """
+    Mounts a DAGsHub repository bucket to a local directory.
+
+    Note: This function is only supported on Linux machines and on macOS via FUSE for macOS (FUSE-T or macFUSE).
+    It may not work as expected on other operating systems due to differences in the handling of filesystem mounts.
+
+    :param repo: The repository in the format "<repo_owner>/<repo_name>". This is used to determine the remote name and
+    mount point.
+    :param cache: Optional. A boolean flag that enables or disables the cache feature. If True, caching is enabled with
+    specific settings.
+    :param path: Optional. A Path object specifying the custom mount path. If not provided, the mount directory is
+    determined based on the current working directory and the repository name.
+    """
+    # Parse the repo string to get the repo owner and name
+    repo_owner, repo_name = repo.split('/')
+
+    # Determine the mount directory
+    if path is not None:
+        mount_dir = Path(path)
+    else:
+        current_dir = Path(os.getcwd()).name
+        if current_dir == repo_name:
+            mount_dir = Path("dagshub_storage")
+        else:
+            mount_dir = Path(repo_name) / "dagshub_storage"
+
+    # 2. Configure the rclone conf
+    remote_name, conf_path = rclone_init(repo_owner=repo_owner, verbose=False)
+    if not conf_path or not conf_path.exists():
+        print("Failed to configure rclone.")
+        return
+
+    # 3. Prepare and execute the mount command
+    try:
+        mount_dir.mkdir(parents=True, exist_ok=True)
+    except OSError:
+        print("Encountered OSError, attempting to remount...")
+        unmount_bucket(repo)
+        mount_dir.mkdir(parents=True, exist_ok=True)
+
+    mount_command = [
+        "rclone", "mount", "--daemon", "--allow-other", "--vfs-cache-mode", "full",
+        f"{remote_name}:{repo_name}/", f"{mount_dir}/"
+    ]
+
+    # Add cache options if needed
+    if cache:
+        mount_command.extend(["--vfs-cache-max-age", "24h"])
+
+    try:
+        # Execute the mount command
+        subprocess.run(mount_command, check=True)
+        print(f"Successfully mounted DagsHub Storage in '{repo_name}' to '{mount_dir}'.")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to mount DagsHub Storage in '{repo_name}': {e}")
+
+
+def unmount_bucket(repo, path=None):
+    """
+    Unmounts a previously mounted DAGsHub repository bucket from the local file system.
```

Comment:
Is there any way in rclone to list current mounts?
Right now this basically mounts and then it's forever there (or until reboot I guess) and the user doesn't really know how to unmount it or where to find what's been done.